src/main/java/org/apache/commons/lang3/math/NumberUtils.java

```diff
@@ -363,44 +366,44 @@ public static Number createNumber(final String str) {
         // and the parsing which will detect if e or E appear in a number due to using the wrong offset
 
         // Detect if the return type has been requested
-        final boolean requestType = !Character.isDigit(lastChar) && lastChar != '.';
-        if (decPos > -1) { // there is a decimal point
-            if (expPos > -1) { // there is an exponent
-                if (expPos <= decPos || expPos > length) { // prevents double exponent causing IOOBE
-                    throw new NumberFormatException(str + " is not a valid number.");
+        final boolean requestType = !Character.isDigit(lastChar) && lastChar != '.'; // + 1
+        if (decPos > -1) { // there is a decimal point  // + 1
+            if (expPos > -1) { // there is an exponent  // + 1
+                if (expPos <= decPos || expPos > length) { // prevents double exponent causing IOOBE // + 2
+                    throw new NumberFormatException(str + " is not a valid number."); // + 1 exit point
                 }
                 dec = str.substring(decPos + 1, expPos);
-            } else {
+            } else {    // + 1
```

Comment:
I don't think "else" should be counted as a decision.

Reply:
But an If-else statement adds two paths so shouldn't it add 2 to the complexity as well?

Reply:
I would say from CFG side, the "if" node still has two out lines, which contribute 2 edges - 1 node = 1 complexity.

Reply:
And if I don't count the elses I am missing like 5 decisions compared to the lizard tool, maybe I missed something?

Reply:
All `if` statements add two branches. One were the if was taken and one were it wasn't. If an `else` exists doesn't really matter.